packages/flutter_tools/lib/src/android/android_workflow.dart

```diff
@@ -58,32 +59,36 @@ class AndroidValidator extends DoctorValidator {
   String _task;
 
   /// Returns false if we cannot determine the Java version or if the version
-  /// is not compatible.
+  /// is older that the minimum allowed version of 1.8.
   Future<bool> _checkJavaVersion(String javaBinary, List<ValidationMessage> messages) async {
     _task = 'Checking Java status';
     try {
       if (!processManager.canRun(javaBinary)) {
         messages.add(ValidationMessage.error(userMessages.androidCantRunJavaBinary(javaBinary)));
         return false;
       }
-      String javaVersion;
+      String javaVersionText;
       try {
         printTrace('java -version');
         final ProcessResult result = await processManager.run(<String>[javaBinary, '-version']);
         if (result.exitCode == 0) {
           final List<String> versionLines = result.stderr.split('\n');
-          javaVersion = versionLines.length >= 2 ? versionLines[1] : versionLines[0];
+          javaVersionText = versionLines.length >= 2 ? versionLines[1] : versionLines[0];
         }
       } catch (error) {
         printTrace(error.toString());
       }
-      if (javaVersion == null) {
+      if (javaVersionText == null) {
         // Could not determine the java version.
         messages.add(ValidationMessage.error(userMessages.androidUnknownJavaVersion));
         return false;
       }
-      messages.add(ValidationMessage(userMessages.androidJavaVersion(javaVersion)));
-      // TODO(johnmccutchan): Validate version.
+      final Version javaVersion = Version.parse(javaVersionText);
+      if (javaVersion < kAndroidJavaMinVersion) {
```

Comment:
Looks like javaVersion can be null here, possibly during a parse failure in Version.parse?

Reply:
OK, I am on it.

Reply:
Hm, it seems that we only check `javaVersionText` for `null` value and not for `isEmpty`, before we parse it.

Reply:
I don't understand why it is still failing, if the `javaVersionText` is null or empty it shouldn't move to parsing step ... unless it gets a text that is not parseable ... 🤔

Reply:
I see the problem now. If you run:
`docker run  gcr.io/flutter-cirrus/build-flutter-image:latest java -version` you will get `openjdk version "1.8.0_212"` which is not parseable by this regex `^(\d+)(\.(\d+)(\.(\d+))?)?`.

I either have to sanitize this string and extract only the version to `Version.parse` or remove the `^` from the current regex.

I lean towards the first, WDYT @jonahwilliams ?